R/windsorize.R

```diff
@@ -1,10 +1,22 @@
 #' Windsorize
-#'
-#' Do some windsorization.
+#' 
+#' Winsorizing or winsorization is the transformation of statistics by limiting
+#' extreme values in the statistical data to reduce the effect of possibly
+#' spurious outliers.
+#' 
+#' @param x A vector.
+#' @param p A quantile.
+#' 
+#' @examples
+#' windsorize(rnorm(5))
+#' 
 #' @export
 windsorize <- function(x, p = .90) {
-  q <- quantile(x, p)
-  x[x >= q] <- q
+  if(all(is.na(x))) {stop("argument should not be a vector containing only NA-s or NULL-s")}
+  q_up <- quantile(x, 0.5 + p / 2 )
```

Comment:
Try to consider the following situations: 

- What happens if `x` contains some `NA` values? 
- What happens if `x` is an empty/NULL vector?